api/config/packages/doctrine.yaml

```diff
@@ -9,12 +9,12 @@ doctrine:
     dbal:
         # configure these for your database server
         driver: 'pdo_pgsql'
-        server_version: '9.6'
+        server_version: '10'
 
         url: '%env(resolve:DATABASE_URL)%'
     orm:
         auto_generate_proxy_classes: true
-        naming_strategy: doctrine.orm.naming_strategy.underscore
+        naming_strategy: App\Doctrine\ORM\Mapping\ResourceNamingStrategy
```

Comment:
Why such change?

Reply:
Because the default and underscore naming strategies both result in `user`, which is a reserved word in SQL.

Reply:
And why doesn't change the table name from the User annotation, instead of creating a new naming_strategy?

Reply:
Because I don't like inconsistencies. But the main reason is, because I already have a nice implementation from previous projects. 😂

Reply:
It'll be an endless debate, but I'm on the camp of pluralized table names. Lol...

Reply:
:+1: for pluralized table names, even if I disagree, you're right it'll be a endless debate without any Symfony/Doctrine convention about it.

Reply:
But I'm not a big fan of adding a class just to rename a table... Some methods in the ResourceNamingStrategy are already in the UnderscoreNamingStrategy. Can you extend or decorate it, instead of rewriting the all class with same behavior?

Reply:
It does more than that actually. The `default` naming strategy (same applies to `underscore`) just doesn't handle many things well.

Reply:
I don't think extending is a good idea. We want full control over how the names are generated.

Reply:
Comparing UnderscoreNamingStrategy::underscore with Inflector::tableize (implemented in your ResourceNamingStrategy): it's the same code. Same for quite all methods in ResourceNamingStrategy.

The only difference I can found is the `preg_replace` delimiter. There is no need to recode an existing class with the same code.

Reply:
Sorry, I don't understand. `UnderscoreNamingStrategy::underscore` is private.

Reply:
And we should not extend just because the implementation is the same. An `extends` relationship `B extends A` is saying that `B is an A`, which clearly does not hold true here.

Reply:
>It does more than that actually. The default naming strategy (same applies to underscore) just doesn't handle many things well.

If we flatten both `UnderscoreNamingStrategy` & `ResourceNamingStrategy` classes with the same structure, the only difference would be the `preg_replace` delimiter. Finally, your class does the same than `UnderscoreNamingStrategy`, I don't see any improvement in it.

Reply:
`ResourceNamingStrategy` <is **not** a> `UnderscoreNamingStrategy`. The implementation details do not matter.